hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/DistributedStream.java

```diff
@@ -60,6 +67,33 @@
 public interface DistributedStream<T> extends Stream<T> {
 
     /**
+     * Returns a distributed {@code Stream} with given processors as its source.
+     */
+    static <T> DistributedStream<T> fromSource(JetInstance instance, ProcessorMetaSupplier metaSupplier) {
+        return new AbstractSourcePipe<T>(new StreamContext(instance)) {
+            @Override
+            protected ProcessorMetaSupplier getSourceMetaSupplier() {
+                return metaSupplier;
+            }
+
+            @Override
+            protected String getName() {
+                return UuidUtil.newUnsecureUuidString();
+            }
+        };
+    }
+
+    /**
+     * Returns a distributed {@code Stream} with given {@code Source} as its source.
+     */
+    static <T> DistributedStream<T> fromSource(JetInstance instance, Source<T> source) {
```

Comment:
I thought about this and let's get rid of this link here. No need to intersect pipeline API with java.util.stream. What do you think?

Reply:
removed it